src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/YFlowReadBolt.java

```diff
@@ -0,0 +1,180 @@
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowhs.bolts;
+
+import static org.openkilda.wfm.topology.utils.KafkaRecordTranslator.FIELD_ID_KEY;
+import static org.openkilda.wfm.topology.utils.KafkaRecordTranslator.FIELD_ID_PAYLOAD;
+
+import org.openkilda.messaging.Message;
+import org.openkilda.messaging.command.CommandData;
+import org.openkilda.messaging.command.yflow.SubFlowsReadRequest;
+import org.openkilda.messaging.command.yflow.SubFlowsResponse;
+import org.openkilda.messaging.command.yflow.YFlowPathsReadRequest;
+import org.openkilda.messaging.command.yflow.YFlowPathsResponse;
+import org.openkilda.messaging.command.yflow.YFlowReadRequest;
+import org.openkilda.messaging.command.yflow.YFlowResponse;
+import org.openkilda.messaging.command.yflow.YFlowsDumpRequest;
+import org.openkilda.messaging.error.ErrorData;
+import org.openkilda.messaging.error.ErrorMessage;
+import org.openkilda.messaging.error.ErrorType;
+import org.openkilda.messaging.error.MessageException;
+import org.openkilda.messaging.info.ChunkedInfoMessage;
+import org.openkilda.messaging.info.InfoData;
+import org.openkilda.messaging.info.InfoMessage;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.wfm.AbstractBolt;
+import org.openkilda.wfm.error.FlowNotFoundException;
+import org.openkilda.wfm.share.zk.ZkStreams;
+import org.openkilda.wfm.share.zk.ZooKeeperBolt;
+import org.openkilda.wfm.topology.flowhs.service.YFlowReadService;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.util.List;
+
+public class YFlowReadBolt extends AbstractBolt {
+    public static final Fields OUTPUT_STREAM_FIELDS = new Fields(FIELD_ID_KEY, FIELD_ID_PAYLOAD);
+
+    private final YFlowReadConfig config;
+
+    private transient YFlowReadService yFlowReadService;
+
+    public YFlowReadBolt(YFlowReadConfig config, PersistenceManager persistenceManager) {
+        super(persistenceManager, config.getLifeCycleEventComponent());
+
+        this.config = config;
+    }
+
+    @Override
+    public void init() {
+        yFlowReadService = new YFlowReadService(persistenceManager,
+                config.getReadOperationRetriesLimit(), config.getReadOperationRetryDelay());
+    }
+
+    protected void handleInput(Tuple input) throws Exception {
+        String requestId = getCommandContext().getCorrelationId();
+        CommandData request = pullValue(input, FIELD_ID_PAYLOAD, CommandData.class);
+
+        try {
+            if (request instanceof YFlowsDumpRequest) {
```

Comment:
Why we are putting read operations into `flowhs`. Should it be into `nbworker`?

Reply:
As per our discussion - all y-flow CRUD functions are to be located in flowhs topology.